tests/capture/test_operators.py

```diff
@@ -0,0 +1,400 @@
+        jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, 0.5)
+
+    assert len(q) == 1
+    assert qml.equal(
+        q.queue[0], qml.Rot(1.0, jax.numpy.sqrt(0.5), 0.25, wires=1), check_interface=False
+    )
+
+
+@pytest.mark.parametrize("as_kwarg", (True, False))
+@pytest.mark.parametrize("w", (0, (0,), [0], range(1), qml.wires.Wires(0)))
+def test_different_wires(w, as_kwarg):
+    def qfunc():
+        if as_kwarg:
+            qml.X(wires=w)
+        else:
+            qml.X(w)
+
+    jaxpr = jax.make_jaxpr(qfunc)()
+
+    assert len(jaxpr.eqns) == 1
+
+    eqn = jaxpr.eqns[0]
+    assert eqn.primitive == qml.X._primitive
+    assert len(eqn.invars) == 1
+    assert isinstance(eqn.invars[0], jax.core.Literal)
+    assert eqn.invars[0].val == 0
+
+    assert isinstance(eqn.outvars[0].aval, AbstractOperator)
+    assert isinstance(eqn.outvars[0], jax.core.DropVar)
+
+    assert eqn.params == {"n_wires": 1}
+
+    with qml.queuing.AnnotatedQueue() as q:
+        jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts)
+
+    assert len(q) == 1
+    assert qml.equal(q.queue[0], qml.X(0))
+
+
+def test_parametrized_op():
+    """Test capturing a parametrized operation."""
+
+    jaxpr = jax.make_jaxpr(qml.Rot)(1.0, 2.0, 3.0, 10)
+
+    assert len(jaxpr.eqns) == 1
+    eqn = jaxpr.eqns[0]
+
+    assert eqn.primitive == qml.Rot._primitive
+    assert len(eqn.invars) == 4
+    assert jaxpr.jaxpr.invars == jaxpr.eqns[0].invars
+
+    assert isinstance(eqn.outvars[0].aval, AbstractOperator)
+    assert eqn.params == {"n_wires": 1}
+
+    with qml.queuing.AnnotatedQueue() as q:
+        jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, 1.0, 2.0, 3.0, 10)
+
+    assert len(q) == 1
+    assert qml.equal(q.queue[0], qml.Rot(1.0, 2.0, 3.0, 10))
+
+
+def test_pauli_rot():
+    """Test a special operation that has positional metadata and overrides binding."""
+
+    def qfunc(a, wire0, wire1):
+        qml.PauliRot(a, "XY", (wire0, wire1))
+
+    jaxpr = jax.make_jaxpr(qfunc)(0.5, 2, 3)
+    assert len(jaxpr.eqns) == 1
+    eqn = jaxpr.eqns[0]
+
+    assert eqn.primitive == qml.PauliRot._primitive
+    assert eqn.params == {"pauli_word": "XY", "id": None, "n_wires": 2}
+
+    assert len(eqn.invars) == 3
+    assert jaxpr.jaxpr.invars == eqn.invars
+
+    with qml.queuing.AnnotatedQueue() as q:
+        jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, 2.5, 3, 4)
+
+    assert len(q) == 1
+    assert qml.equal(q.queue[0], qml.PauliRot(2.5, "XY", (3, 4)))
+
+
+class TestTemplates:
+    def test_variable_wire_non_parametrized_template(self):
+        """Test capturing a variable wire, non-parametrized template like GroverOperator."""
+
+        jaxpr = jax.make_jaxpr(qml.GroverOperator)(wires=(0, 1, 2, 3, 4, 5))
+
+        assert len(jaxpr.eqns) == 1
+        eqn = jaxpr.eqns[0]
+
+        assert eqn.primitive == qml.GroverOperator._primitive
+        assert eqn.params == {"n_wires": 6}
+        assert eqn.invars == jaxpr.jaxpr.invars
```

Comment:
Could test output like above

```suggestion
        assert eqn.invars == jaxpr.jaxpr.invars
        assert isinstance(eqn.outvars[0].aval, AbstractOperator)
```